n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,134 @@
+from ekobox import evol_pdf, gen_theory, gen_op, genpdf, gen_info
+import pathlib
+import numpy as np
+import yaml
+from . import utils
+from validphys.loader import Loader
+import os
+from ekomark import apply
+from eko import basis_rotation as br
+
+
+def evolve_fit(conf_folder):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit 
+    """
+    initial_PDFs_dict = load_fit(conf_folder)
+    eko, theory, op = construct_eko_for_fit(conf_folder)
+    for replica in initial_PDFs_dict.keys():
+        evolve_exportgrid(
+            initial_PDFs_dict[replica],
+            eko,
+            theory,
+            op,
+            int(replica.removeprefix("replica_")),
+            conf_folder,
+        )
+
+
+def load_fit(conf_folder):
+    """
+    Loads all the replica pdfs at fitting scale in conf_folder and return the exportgrids
+
+    Parameters
+    ----------
+
+        conf_folder: str
+            path to the folder containing the fit  
+    
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = pathlib.Path(conf_folder) / "nnfit"
+    replica_list = []
+    for subdir, dirs, files in os.walk(nnfitpath):
+        for dir in dirs:
+            replica_list.append(dir)
+    replica_list.remove("input")
+    pdf_dict = {}
+    for replica in replica_list:
+        rep_path = pathlib.Path(replica) / (conf_folder + ".exportgrid")
+        yaml_file = nnfitpath / rep_path.relative_to(rep_path.anchor)
+        with yaml_file.open() as fp:
+            data = yaml.safe_load(fp)
+        pdf_dict[replica] = data
+    return pdf_dict
+
+
+# Temporary solution. Then it will be loaded from the theory itself
+def construct_eko_for_fit(conf_folder):
+    """
+    Construct the eko operator needed for evolution of fitted pdfs
+
+    Parameters
+    ----------
+        conf_folder: str
+            path to the folder containing the fit  
+    """
+    # read the runcard
+    my_runcard = utils.read_runcard(conf_folder)
+    # theory_card construction
+    theory = Loader().check_theoryID(my_runcard["theory"]["theoryid"]).get_description()
+    theory.pop("FNS")
```

Comment:
`FNS` is simply ignored in `eko` so you can pop it or not ...

Reply:
I believe this is what we are doing also in ekobox, right?